src/components/views/rooms/BasicMessageComposer.tsx

```diff
@@ -536,6 +537,17 @@ export default class BasicMessageEditor extends React.Component<IProps, IState>
             }
         }
 
+        const navAction = getKeyBindingsManager().getNavigationAction(event);
+        switch (navAction) {
+            case KeyBindingAction.PreviousLandmark:
+                LandmarkNavigation.navigateToPreviousLandmarkFrom(Landmark.MESSAGE_COMPOSER_OR_HOME);
+                handled = true;
+                break;
+            case KeyBindingAction.NextLandmark:
+                LandmarkNavigation.navigateToNextLandmarkFrom(Landmark.MESSAGE_COMPOSER_OR_HOME);
+                handled = true;
+                break;
```

Comment:
a `findAndFocusNextLandmark(enumValue, navAction === KeyBinding.PreviousLandmark)` would simplify this boilerplate in every receiver class

Reply:
https://github.com/matrix-org/matrix-react-sdk/pull/12190/commits/ea4fea8b96af0c010ce4598b04459d7bc3046549